### PR TITLE
WASM: Fix System.Diagnostics.TextWriterTraceListener tests

### PR DESCRIPTION
**Pull request by akoeplinger**
It was using Process.GetCurrentProcess() which throws PNSE on WebAssembly.

### CODE REVIEW

File: src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs [hunk 1]
```diff
@@ -258,8 +258,15 @@ private void WriteEndHeader()
...
             if (processName is null)
             {
-                using Process process = Process.GetCurrentProcess();
-                s_processName = processName = process.ProcessName;
+                try
+                {
+                    using Process process = Process.GetCurrentProcess();
+                    s_processName = processName = process.ProcessName;
+                }
+                catch (PlatformNotSupportedException)
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs:266]:
```suggestion
                catch (PlatformNotSupportedException) // Process isn't supported on Browser
```

Comment (stephentoub) [reply src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs:266]: Did you consider instead doing a platform check?
```C#
if (RuntimeInformation.IsOSPlatform(OSPlatform.Browser) { ... }
else { ... }
```

Comment (akoeplinger) [reply src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs:266]: yeah but I think over the last couple PRs we landed on the consensus to prefer catching PNSE instead.

Comment (stephentoub) [reply src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs:266]: > over the last couple PRs

Oh there have been more of these?  I must have missed that.  Could you point me to some examples?

Comment (akoeplinger) [reply src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs:266]: E.g. https://github.com/dotnet/runtime/commit/b62b16acf506f4e5c0d8b4ebaf18875ac9ad3477#diff-ce3739bb88265fab85205217d762f650R70-R79

Comment (stephentoub) [reply src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs]: Ok.

**Review by safern — APPROVED** [review]

**Pull request merged**